src/Integrations/Integrations/Swoole/SwooleIntegration.php

```diff
@@ -0,0 +1,202 @@
+                $integration->addTraceAnalyticsIfEnabled($rootSpan);
+
+                $args = $hook->args;
+                /** @var Request $request */
+                $request = $args[0];
+
+                $headers = [];
+                $allowedHeaders = \dd_trace_env_config('DD_TRACE_HEADER_TAGS');
+                foreach ($request->header as $name => $value) {
+                    $headers[strtolower($name)] = $value;
+                    $normalizedHeader = preg_replace("([^a-z0-9-])", "_", strtolower($name));
+                    if (\array_key_exists($normalizedHeader, $allowedHeaders)) {
+                        $rootSpan->meta["http.request.headers.$normalizedHeader"] = $value;
+                    }
+                }
+                consume_distributed_tracing_headers(function ($key) use ($headers) {
+                    return $headers[$key] ?? null;
+                });
+
+                if (\dd_trace_env_config("DD_TRACE_CLIENT_IP_ENABLED")) {
+                    $res = extract_ip_from_headers($headers + ['REMOTE_ADDR' => $request->server['remote_addr']]);
+                    $rootSpan->meta += $res;
+                }
+
+                if (isset($headers["user-agent"])) {
+                    $rootSpan->meta["http.useragent"] = $headers["user-agent"];
+                }
+
+                $rawContent = $request->rawContent();
+                if ($rawContent) {
+                    // The raw content will always be populated if the request is a POST request, independent of the
+                    // Content-Type header.
+                    // However, it may not be json-decodable
+                    $postFields = json_decode($rawContent, true);
+                    if (is_null($postFields)) {
+                        // Fallback to the post fields, which is an array
+                        // This array is not always populated, depending on the Content-Type header
+                        $postFields = $request->post;
+                    }
+                }
+                if (!empty($postFields)) {
+                    $postFields = Normalizer::sanitizePostFields($postFields);
+                    foreach ($postFields as $key => $value) {
+                        $rootSpan->meta["http.request.post.$key"] = $value;
+                    }
+                }
+
+                $normalizedPath = Normalizer::uriNormalizeincomingPath(
+                    $request->server['request_uri']
+                    ?? $request->server['path_info']
+                    ?? '/'
+                );
+                $rootSpan->resource = $request->server['request_method'] . ' ' . $normalizedPath;
+                $rootSpan->meta[Tag::HTTP_METHOD] = $request->server['request_method'];
+
+                $serverProtocol = $request->server['server_protocol'] ?? 'HTTP/1.1';
+                $scheme = strpos($serverProtocol, 'HTTPS') !== false ? 'https://' : 'http://';
```

Comment:
really? TLS should be independent from the protocol, unless that's something special swoole does?

Reply:
That's wrong

Reply:
Changed this part to use the server's SSL property. Since I seemingly don't have access to the URL, I think that's the best way of fingering the scheme

Reply:
Yes, checking ssl vs non-ssl is the right way to do that.